source/CsvBenchmark/CsvReaderBenchmarks.cs

```diff
@@ -201,12 +201,29 @@ public void MgholamFastCSV()
 
 		class CursivelyStringVisitor : CsvReaderVisitorBase
 		{
+			readonly bool doPooling;
+			readonly byte[] bytes = new byte[1024];
+			int bytesUsed = 0;
+
 			// in any realistic scenario we'd need to at least know the column oridnal to do anything with the record
 			int ordinal = 0;
 
+			public CursivelyStringVisitor(bool doPooling)
+			{
+				this.doPooling = doPooling;
+			}
+
 			public override void VisitEndOfField(System.ReadOnlySpan<byte> chunk)
 			{
-				var str = Encoding.UTF8.GetString(chunk);
+				if (bytesUsed != 0)
+				{
+					chunk.CopyTo(bytes.AsSpan(bytesUsed, chunk.Length));
+					chunk = new ReadOnlySpan<byte>(bytes, 0, bytesUsed + chunk.Length);
+					bytesUsed = 0;
+				}
+				var str = chunk.Length == 1 && chunk[0] < 128 && doPooling
```

Comment:
Move "doPooling" as the first check in the chain. No reason to check the other constraints if pooling is disabled. I measured this to make a measurable difference in the pooling=false case.

Reply:
Updated, thanks.